worlds/kh1/__init__.py

```diff
@@ -0,0 +1,300 @@
+from .Items import KH1Item, KH1ItemData, event_item_table, get_items_by_category, item_table, item_name_groups
+from .Locations import KH1Location, location_table, get_locations_by_category, location_name_groups
+from .Options import KH1Options, kh1_option_groups
+from .Regions import create_regions
+from .Rules import set_rules
+from .Presets import kh1_option_presets
+from worlds.LauncherComponents import Component, components, Type, launch_subprocess
+
+
+def launch_client():
+    from .Client import launch
+    launch_subprocess(launch, name="KH1 Client")
+
+
+components.append(Component("KH1 Client", "KH1Client", func=launch_client, component_type=Type.CLIENT))
+
+
+class KH1Web(WebWorld):
+    theme = "ocean"
+    tutorials = [Tutorial(
+            "Multiworld Setup Guide",
+            "A guide to setting up the Kingdom Hearts Randomizer software on your computer."
+            "This guide covers single-player, multiworld, and related software.",
+            "English",
+            "kh1_en.md",
+            "kh1/en",
+            ["Gicu"]
+    )]
+    option_groups = kh1_option_groups
+    options_presets = kh1_option_presets
+
+
+class KH1World(World):
+    """
+    Kingdom Hearts is an action RPG following Sora on his journey 
+    through many worlds to find Riku and Kairi.
+    """
+    game = "Kingdom Hearts"
+    options_dataclass = KH1Options
+    options: KH1Options
+    topology_present = True
+    required_client_version = (0, 3, 5)
+    web = KH1Web()
+
+    item_name_to_id = {name: data.code for name, data in item_table.items()}
+    location_name_to_id = {name: data.code for name, data in location_table.items()}
+    item_name_groups = item_name_groups
+    location_name_groups = location_name_groups
+    fillers = {}
+    fillers.update(get_items_by_category("Item"))
+    fillers.update(get_items_by_category("Camping"))
+    fillers.update(get_items_by_category("Stat Ups"))
+
+    def create_items(self):
+        #Handle starting worlds
+        starting_worlds = []
+        if self.options.starting_worlds > 0:
+            possible_starting_worlds = ["Wonderland", "Olympus Coliseum", "Deep Jungle", "Agrabah", "Monstro", "Halloween Town", "Neverland", "Hollow Bastion"]
+            if self.options.atlantica:
+                possible_starting_worlds.append("Atlantica")
+            if self.options.end_of_the_world_unlock == "item":
+                possible_starting_worlds.append("End of the World")
+            starting_worlds = self.random.sample(possible_starting_worlds, min(self.options.starting_worlds.value, len(possible_starting_worlds)))
+            for starting_world in starting_worlds:
+                self.multiworld.push_precollected(self.create_item(starting_world))
+        
+        item_pool: List[KH1Item] = []
+        possible_level_up_item_pool = []
+        level_up_item_pool = []
+        
+        #Calculate Level Up Items
+        if True: #Allow notepad++ to collapse this section
```

Comment:
Bruh lmao

I mean it's ok, but it gave me a good laugh

Reply:
Although using an actual IDE like PyCharm would show you a lot of the code style issues that this world has. For being coded in NotePad++, the amount of violations is actually pretty low, so I'm somewhat impressed, but some of the choices are... a little cursed for sure

Reply:
*comment*